tiledb/sm/filesystem/filesystem_base.cc

```diff
@@ -38,8 +38,8 @@ namespace tiledb::sm {
 /*                API                */
 /* ********************************* */
 
-bool FilesystemBase::supports_uri(const URI& uri) const {
-  throw UnsupportedURI(uri.to_string());
+bool FilesystemBase::operator==(const FilesystemBase& fs) const {
+  return typeid(this) == typeid(fs);
 }
```

Comment:
Is this necessary? When doing `if (&old_fs == &new_fs)`, we compare pointers and not values, which should always be possible, shouldn't it?